packages/openapi3/test/examples.test.ts

```diff
@@ -234,4 +234,384 @@ worksFor(["3.0.0", "3.1.0"], ({ openApiFor }) => {
+        desc: "label (object) explode: false",
+        route: "/{.color}",
+        param: `@path color: Record<int32>`,
+        paramExample: `#{R: 100, G: 200, B: 150}`,
+        expectedExample: ".R,100,G,200,B,150",
+      },
+      {
+        desc: "label (object) explode: true",
+        route: "/{.color*}",
+        param: `@path color: Record<int32>`,
+        paramExample: `#{R: 100, G: 200, B: 150}`,
+        expectedExample: ".R=100.G=200.B=150",
+      },
+    ])(
+      "set example on the path parameter with style $desc",
+      async ({ param, route, paramExample, expectedExample }) => {
+        const res: OpenAPI3Document = await openApiFor(
+          `
+          @opExample(#{
+            parameters: #{
+              color: ${paramExample},
+            },
+          })
+          @route("${route}")
+          op getColors(${param}): void;
+          `,
+        );
+        expect((res.paths[`/{color}`].get?.parameters[0] as OpenAPI3Parameter).example).toEqual(
+          expectedExample,
+        );
+      },
+    );
+
+    it.each([
+      {
+        desc: "simple (undefined)",
+        param: `@header color: string | null`,
+        paramExample: `null`,
+        expectedExample: "",
+      },
+      {
+        desc: "simple (string)",
+        param: `@header color: string`,
+        paramExample: `"blue"`,
+        expectedExample: "blue",
+      },
+      {
+        desc: "simple (array) explode: false",
+        param: `@header color: string[]`,
+        paramExample: `#["blue", "black", "brown"]`,
+        expectedExample: "blue,black,brown",
+      },
+      {
+        desc: "simple (array) explode: true",
+        param: `@header(#{ explode: true }) color: string[]`,
+        paramExample: `#["blue", "black", "brown"]`,
+        expectedExample: "blue,black,brown",
+      },
+      {
+        desc: "simple (object) explode: false",
+        param: `@header color: Record<int32>`,
+        paramExample: `#{R: 100, G: 200, B: 150}`,
+        expectedExample: "R,100,G,200,B,150",
+      },
+      {
+        desc: "simple (object) explode: true",
+        param: `@header(#{ explode: true }) color: Record<int32>`,
+        paramExample: `#{R: 100, G: 200, B: 150}`,
+        expectedExample: "R=100,G=200,B=150",
+      },
+    ])(
+      "set example on the header parameter with style $desc",
+      async ({ param, paramExample, expectedExample }) => {
+        const res: OpenAPI3Document = await openApiFor(
+          `
+          @opExample(#{
+            parameters: #{
+              color: ${paramExample},
+            },
+          })
+          @route("/")
+          op getColors(${param}): void;
+          `,
+        );
+        expect((res.paths[`/`].get?.parameters[0] as OpenAPI3Parameter).example).toEqual(
+          expectedExample,
+        );
+      },
+    );
+
+    it.each([
+      {
+        desc: "form (undefined)",
+        param: `@cookie color: string | null`,
+        paramExample: `null`,
```

Comment:
shouldn't this test case be not passing the example value?

otherwise for null i feel like it probably still serialize as `color=null` (same for all `(undefined)`) cases

Reply:
There's one more column to the tables, `undefined`, that I didn't include in the comments.
> The undefined column replaces the empty column in previous versions of this specification in order to better align with [[RFC6570](https://spec.openapis.org/oas/v3.0.4.html#bib-rfc6570)] [Section 2.3](https://datatracker.ietf.org/doc/html/rfc6570#section-2.3) terminology, which describes certain values including but not limited to null as “undefined” values with special handling; notably, the empty string is not undefined

So I took the value from the `undefined` columns for these cases.
(link to table: https://spec.openapis.org/oas/v3.0.4.html#style-examples)